Rewritten add_path and pylint corrections

OneTorcha opened the pull request.

OneTorcha committed:
rfcli.py:
- pylint: Your code has been rated at 10.00/10 - as much python3 changes as noticed.
- removed add_path() and replaced it with code from robotframework 6.1.x so that rfcli works pre 6.1 versions.

OneTorcha committed:
rfcli.py:
- pylint: Your code has been rated at 10.00/10 - as much python3 changes as noticed.
- removed add_path() and replaced it with code from robotframework 6.1.x so that rfcli works pre 6.1 versions.

OneTorcha committed:
rfcli.py: code cleanup

OneTorcha committed:
Merge remote-tracking branch 'origin/rewritten_load_and_pylint_corrections' into rewritten_load_and_pylint_corrections

OneTorcha committed:
rfcli.py:
- removed check if robot is in sys.modules as useless.

vereszol commented on src/crl/rfcli/rfcli.py:
```diff
@@ -201,8 +198,6 @@ def use_jybot(self):
             if RobotCommand.is_jybot_installed():
                 return True
-            else:
-                raise Exception("Jybot is not installed.")
```
Could you explain the reason to remove this?

OneTorcha replied:
Misunderstood the pylint. 

 ```
def use_jybot(self):
        if self.rfcli_args.enable_jybot:
            if RobotCommand.is_jybot_installed():
                return True
            raise ValueError("Jybot is not installed.")
        return False
```

vereszol commented on src/crl/rfcli/rfcli.py:
```diff
@@ -223,8 +218,8 @@ def run(self):
-                print('HTML logs might be located at: http://%s/~%s/rfcli/log.html' % (
-                    socket.getfqdn(), self._user_dir_property()))
+                print(f"HTML logs might be located at: http://%s/~%s/rfcli/log.html"
+                      f" {socket.getfqdn(), self._user_dir_property()}")
```
The resulting printout is not the same as before.

OneTorcha replied:
print(f"HTML logs might be located at: http://{socket.getfqdn()}/~{self._user_dir_property()}/rfcli/log.html")

vereszol commented on src/crl/rfcli/rfcli.py:
```diff
@@ -268,27 +262,35 @@ def _get_environment(self):
-        except Exception as e:
-            raise Exception("Cannot open target ini file %s: %s" % (self.absfilename, e))
-        if not self.parser.has_section('target'):
-            raise Exception("Target ini file %s does not have the [target] section" % self.absfilename)
```
I think the same checks should be done and same exceptions should be raised (or some subclass of Exception)

OneTorcha replied:
I get same errors without those exections:

robot run-test: commands[1] | rfcli --statusrc --target testi9 -i koe testcases/entity-test/file_hunter/file_hunter.robot
/work/targets/testi9.ini
NoSectionError: No section: 'target'

robot run-test: commands[1] | rfcli --statusrc --target test9 -i koe testcases/entity-test/file_hunter/file_hunter.robot
ValueError: Target file test9 does not exist

vereszol replied:
For a human user these might be the same errors, but programmatically they are different.

Exception type change (NoSectionError and ValueError vs the previous Exception) are not a problem as the new ones are subclass of the old, but the change in exception message means backwards incompatible interface change for anyone using this library programmatically.

This is not a critical problem - and I understand it simplifies the code - but it should be released as version 2.0.0 as instructed in CRL workflow at https://crl-devutils.readthedocs.io/en/latest/workflow.html?highlight=version#is-version-updated.

OneTorcha replied:
Ok, I bring those errors back if we figure out the path issue. Small but might cause some projects a lot of editing :)

OneTorcha committed:
rfcli.py: PR comments fix:
- use_jybot was missing raise valueerror
- run() had error in printout

theodore86 commented:
@OneTorcha do you need help on this one? Looking forward to get the latest release

OneTorcha commented:
@theodore86 maybe, review could be nice and for some reason sourcing libs and resources need "./" after this change..

like libraries/own_lib.py needs to be modified ../libraries/own_lib.py

Just havent had time to figure out or is that a big problem even.

theodore86 commented:
@OneTorcha thanks, i will get a look at it

theodore86 commented:
```console
rfcli -t targets/target9.ini test.robot 
/tmp/targets/target9.ini
==============================================================================
Test                                                                          
==============================================================================
Test 01                                                               test
Test 01                                                               | PASS |
------------------------------------------------------------------------------
Test                                                                  | PASS |
1 test, 1 passed, 0 failed
==============================================================================
Debug:   /tmp/rfcli_output/debug.txt
Output:  /tmp/rfcli_output/output.xml
Log:     /tmp/rfcli_output/log.html
Report:  /tmp/rfcli_output/report.html
(⎈|:N/A)vagrant@sandbox:/tmp$ rfcli -t target9 test.robot 
/tmp/targets/target9.ini
==============================================================================
Test                                                                          
==============================================================================
Test 01                                                               test
Test 01                                                               | PASS |
------------------------------------------------------------------------------
Test                                                                  | PASS |
1 test, 1 passed, 0 failed
==============================================================================
Debug:   /tmp/rfcli_output/debug.txt
Output:  /tmp/rfcli_output/output.xml
Log:     /tmp/rfcli_output/log.html
Report:  /tmp/rfcli_output/report.html
(⎈|:N/A)vagrant@sandbox:/tmp$ cat targets/target9.ini 
[target]
USERNAME=test
```

theodore86 commented:
> @theodore86 maybe, review could be nice and for some reason sourcing libs and resources need "./" after this change..
> 
> like libraries/own_lib.py needs to be modified ../libraries/own_lib.py
> 
> Just havent had time to figure out or is that a big problem even.

Well it seems that you have remove the functionality which sets the libraries, testcases and resources to ```PYTHONPATH```:

```python
            print("Environment: %s" % self.full_environment)
            print("Commandline: %s" % self.commandline)
        # pylint: disable=unused-variable
        for key, value in self.new_environment_variables.items():
            newpaths = value.split(os.pathsep)
            for path in newpaths:
                pythonpathsetter.add_path(path, True)
```

derives from:

```python
    def run(self):
        if self.help_only:
            self.parser.print_help()
            return 0
        if self.use_jybot():
            command = JybotCommand(self.get_robot_options(), self.robot_args, self.get_environment(),
                                   debug=self.rfcli_args.debug)
        else:
            command = PybotCommand(self.get_robot_options(), self.robot_args, self.get_environment(),
                                   debug=self.rfcli_args.debug)
```

```python
    def get_environment(self):
        return {} if self.rfcli_args.no_pythonpath else self._get_environment()

    def _get_environment(self):
        testcase_libs = [
            i[0] for i in os.walk(os.path.join(self.calldir, 'testcases'), followlinks=True)
            if re.search(os.path.join('.*', 'libraries$'), i[0])]
        path_list = [self.calldir, os.path.join(
            self.calldir, 'libraries'), os.path.join(
                self.calldir, 'resources')] + testcase_libs
        return {'PYTHONPATH': os.pathsep.join(path_list)}

````

The proposal is this one (restore the part that you have removed and replace the following lines):

```python
        for key, value in self.new_environment_variables.items():
            newpaths = value.split(os.pathsep)
            for path in newpaths:
                if not any(fnmatch.fnmatch(p, path) for p in sys.path):
                    sys.path.append(path)
```

You just need 2 lines of code (```import fnmatch```)

OneTorcha commented:
add_path() has been removed from 6.1.1, hence this "hack"/"fix". And I was trying to get it work bot 6.1.1 and older versions. I'll try your proposal, looks good.

theodore86 commented:
> add_path() has been removed from 6.1.1, hence this "hack"/"fix". And I was trying to get it work bot 6.1.1 and older versions. I'll try your proposal, looks good.

```robot.pythonpathsetter``` is not part of robot.api, indeed seems like a "hack" but adding those 2 lines of code (check the code above) and removing the ```pythonpathsetter``` you are done, the original implementation "appends" each path in the ```sys.path``` if not exists.

OneTorcha commented:
So its better to make new branch and make just those little changes so I wont mess everything up :)

theodore86 commented:
> ```python
> if not any(fnmatch.fnmatch(p, path) for p in sys.path):
>                     sys.path.append(path)
> ```

Yea those 2 lines shall do the work.

It works btw:) waiting for the new release!

OneTorcha committed:
rfcli.py: PR comments fix, part 2:
- execute() modified as proposed, importing fnmatch and adding to path.
- all old excpetions are back.

OneTorcha commented:
@vereszol @theodore86 There is  PR comments fix, part 2. I think I covered everything, if not, ups :)

theodore86 commented:
@OneTorcha do you need any assistance on the PR comments?

OneTorcha commented:
> @OneTorcha do you need any assistance on the PR comments?

Hi, what you mean? Do I have to Close With comment all? :)

theodore86 commented:
> > @OneTorcha do you need any assistance on the PR comments?
> 
> Hi, what you mean? Do I have to Close With comment all? :)

When are you planning to release the rfcli?

OneTorcha commented:
> > > @OneTorcha do you need any assistance on the PR comments?
> > 
> > 
> > Hi, what you mean? Do I have to Close With comment all? :)
> 
> When are you planning to release the rfcli?

Whenever @vereszol accepts this and pulls the trigger.

OneTorcha committed:
rfcli.py: yaml.load was overwritten with old version.

OneTorcha committed:
rfcli.py: yaml.load was overwritten with old version. safe_load was the correct one.

theodore86 commented:
> > > > @OneTorcha do you need any assistance on the PR comments?
> > > 
> > > 
> > > Hi, what you mean? Do I have to Close With comment all? :)
> > 
> > 
> > When are you planning to release the rfcli?
> 
> Whenever @vereszol accepts this and pulls the trigger.

Do we need someone else to approve this PR?

OneTorcha commented:
> > > > > @OneTorcha do you need any assistance on the PR comments?
> > > > 
> > > > 
> > > > Hi, what you mean? Do I have to Close With comment all? :)
> > > 
> > > 
> > > When are you planning to release the rfcli?
> > 
> > 
> > Whenever @vereszol accepts this and pulls the trigger.
> 
> Do we need someone else to approve this PR?

Smaller commits (one huge was not inclusive enough :) ), some refactoring for CI stuff.

theodore86 commented:
Whats the issue with the PR? It takes ages to be merged, The maintainers are not aware of it?

OneTorcha commented:
> Whats the issue with the PR? It takes ages to be merged, The maintainers are not aware of it?

Too many changes in one PR. Pipe is porked. Nobody cares. I use the fix daily with my own hack and I have done what I need to do so rest is on that merge pull request button :)
